packages/devtools_app/lib/src/error_badge_manager.dart

```diff
@@ -144,21 +138,33 @@ class ErrorBadgeManager extends DisposableController
   void filterErrors(String screenId, bool Function(String id) isValid) {
     final errors = _activeErrors[screenId];
     if (errors == null) return;
-    errors.value =
+
+    final oldCount = errors.value.length;
+    final newValue =
         Map.fromEntries(errors.value.entries.where((e) => isValid(e.key)));
-    _updateErrorCount(screenId);
+    if (newValue.length != oldCount) {
+      errors.value = newValue;
+    }
   }
 
   void markErrorAsRead(String screenId, DevToolsError error) {
+    clearErrors(screenId);
```

Comment:
do we need to do this here? shouldn't this be the responsibility of the screen?

Reply:
This was an error - it was already being called before this function. I've removed in 04143cd0!